weldx/asdf/validators.py

```diff
@@ -516,120 +517,75 @@ def _custom_shape_validator(
     return dict_values
 
 
-def wx_unit_validator(
-    validator, wx_unit, instance, schema
-) -> Iterator[ValidationError]:
-    """Custom validator for checking dimensions for objects with 'unit' property.
+# VALIDATOR CLASSES --------------------------------------------------------------------
 
-    ASDF documentation:
-    https://asdf.readthedocs.io/en/2.8.1/asdf/extensions.html#adding-custom-validators
 
-    Parameters
-    ----------
-    validator:
-        A jsonschema.Validator instance.
-    wx_unit:
-        Enable unit validation for this schema.
-    instance:
-        Tree serialization (with default dtypes) of the instance
-    schema:
-        Dict representing the full ASDF schema.
+class WxUnitValidator(Validator):
+    """Custom validator for checking dimensions for objects with 'unit' property."""
 
-    Yields
-    ------
-    asdf.ValidationError
+    schema_property = "wx_unit"
+    tags = ["**"]
```

Comment:
Ideally we would only list `weldx` tags here but selecting the validator for every node preserves the legacy behavior for *untagged* nodes
see previous discussion here https://github.com/asdf-format/asdf/pull/1050#issuecomment-1027373230